cilium/CFP-033-move-hubble-into-cilium.md

```diff
@@ -0,0 +1,120 @@
+- Robin Hahling (`@rolinh`)
+- Anna Kapuścińska (`@lambdanis`)
+- Tobias Klauser (`@tklauser`)
+- Tam Mach (`@sayboras`)
+- Alexandre Perrin (`@kaworu`)
+- Liz Rice (`@lizrice`)
+- Glib Smaga (`@glibsm`)
+- Joe Stringer (`@joestringer`)
+- Sebastian Wicki (`@gandro`)
+- Chance Zibolski (`@chancez`)
+
+## Summary
+
+Move `cilium/hubble` code into `cilium/cilium` repo to improve the Hubble CLI
+development and release processes.
+
+## Motivation
+
+Currently, Hubble server code lives in `cilium/cilium` repo, while Hubble CLI
+code is in a separate `cilium/hubble` repo. This complicates the Hubble CLI
+development and release processes.
+
+### Development process
+
+Currently, modifying Hubble APIs requires dealing with multiple repos:
+
+- Make API and server-side changes in cilium/cilium repo.
+- Update go.mod to pick up the cilium/cilium changes in cilium/hubble repo, and
+  make corresponding client-side changes.
+- Wait for the next Hubble CLI release.
+- Wait for Renovate to update [download-hubble.sh](https://github.com/cilium/cilium/blob/main/images/cilium/download-hubble.sh)
+  before you can use the new Hubble CLI from the `cilium-agent` container.
+
+### Release process
+
+The current Hubble CLI release process requires:
+
+- Vendor unreleased latest `cilium/cilium` code in `cilium/hubble` repo before a
+  new `cilium/cilium` minor version gets released to pick up new Hubble API
+  changes.
+- Release a new Hubble CLI.
+- Make sure [download-hubble.sh](https://github.com/cilium/cilium/blob/main/images/cilium/download-hubble.sh)
+  gets updated before the corresponding `cilium/cilium` minor version gets
+  released.
+ 
+## Goals
+
+The goal of this proposal is to simplify the Hubble CLI development / release
+processes.
+
+* Enable developers to make Hubble server and CLI changes in a single pull
+  request, and immediately make the new Hubble CLI available in the CI
+  `cilium-agent` images without having to wait for the next Hubble CLI release.
+* Build Hubble CLI release binaries from Cilium release tags to simplify the
+  release process.
+
+## Non-Goals
+
+* Archive or delete the cilium/hubble repo.
+
+## Proposal
+
+This proposal consists of two phases.
+
+### Phase 1: Move cilium/hubble code to cilium/cilium repo
+
+In the fist phase, move `cilium/hubble` code to `cilium/cilium` repo without
+changing the current Hubble CLI release process. See the following draft
+pull requests for more details regarding the first phase:
+
+- https://github.com/cilium/cilium/pull/31893
+- https://github.com/cilium/hubble/pull/1444
+
+### Phase 2: Update the release process
+
+In the second phase, coordinate with `@cilium/sig-hubble` and update the Hubble
+CLI release process. Instead of having a separate GitHub workflow in
+`cilium/hubble` repository, pushing a release tag in `cilium/cilium` repo can
+trigger a workflow to build Hubble CLI release binaries and publish a release
+in `cilium/hubble` repo.
+
+## Impacts / Key Questions
```

Comment:
I think another impact of Phase I is that users importing `cilium/hubble` as a go module will now have to vendor in Cilium. Impact seems rather low, mainly affecting cilium/cilium-cli, cilium/fake, cilium/hubble-ui and the already archived hubble-otel:

https://pkg.go.dev/github.com/cilium/hubble@v0.13.2/pkg/printer?tab=importedby
https://pkg.go.dev/github.com/cilium/hubble@v0.13.2/pkg/time?tab=importedby
https://pkg.go.dev/github.com/cilium/hubble@v0.13.2/pkg/defaults?tab=importedby

The only notable known import which is not a fork and not a Cilium org member is https://github.com/stleox/seeflow

Reply:
Another impacted user is the Homebrew recipe: https://github.com/Homebrew/homebrew-core/blob/master/Formula/h/hubble.rb

Reply:
yeah i was just thinking of mentioning the homebrew stuff. i'll update the CFP 📝